Typescript throwing errors on build when building a component lib using Panda

### Description

When attempting to run a build for a component library using Panda, typescript is throwing the following errors: 
```
src/components/Button/index.tsx(10,14): error TS2742: The inferred type of 'Button' cannot be named without a reference to '../../../node_modules/@acmeorg/styled-system/types/jsx'. This is likely not portable. A type annotation is necessary.
src/preset/recipes/button.ts(3,14): error TS2742: The inferred type of 'button' cannot be named without a reference to '../../../node_modules/@acmeorg/styled-system/types/recipe'. This is likely not portable. A type annotation is necessary.
```

Though, this is not limited to `/types/recipe` and `/types/jsx` I have enountered it for many of the `types` subdirectories.

### Link to Reproduction

https://stackblitz.com/edit/vitejs-vite-nyg6dl

### Steps to reproduce

Have a similar setup to the one provided in the stackblitz link, and run the build.
(Sorry not really steps, but the example should cover it)

### JS Framework

React (TS)

### Panda CSS Version

0.13.1

### Browser

N/A - Occurs at build time

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

I had brought this up in the discord and thought I could just use a `ReturnType<typeof styled>` kind of fix but this just led to other issues so is no longer an option.

I can confirm that adding additional exports to `styled-system/types/index.d.ts` does fix the issue. Though I am not certain on which exports exactly are required, and which might not be.

--- a/.changeset/chilly-jobs-try.md
+++ b/.changeset/chilly-jobs-try.md
@@ -1,0 +1,6 @@
+---
+"@pandacss/dev": patch
+---
+
+Add missing types (PatternConfig, RecipeConfig, RecipeVariantRecord) to solve a TypeScript issue (The inferred type of
+xxx cannot be named without a reference...)

--- a/packages/cli/src/index.ts
+++ b/packages/cli/src/index.ts
@@ -104,6 +104,8 @@ export type {
   LayerStyles,
   Preset,
   PropertyConfig,
+  RecipeConfig,
+  RecipeVariantRecord,
   SemanticTokens,
   SlotRecipeConfig,
   SlotRecipeVariantRecord,

--- a/packages/cli/src/index.ts
+++ b/packages/cli/src/index.ts
@@ -106,6 +106,7 @@ export type {
   PropertyConfig,
   RecipeConfig,
   RecipeVariantRecord,
+  PatternConfig,
   SemanticTokens,
   SlotRecipeConfig,
   SlotRecipeVariantRecord,

